egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -347,4 +347,164 @@ def AddClstmLayer(config_lines,
             }
 
 
+def AddBlstmLayer(config_lines,
```

Comment:
Oh..those weren't comments for the current LSTM implementation. They were
from an older version of LSTM. I was just suggesting that you write
equations like these at the top of the BLSTM just to make people understand
what you are doing.

I would suggest having description which consists of
-- Actual mathematical equations
-- Implementation level equations .. describing the way you are computing
things.. e.g. if you combining some matrix multiplications
-- and then the actual config code

This helps people understand and modify things easily. I have been a bit
lazy in doing this for the LSTM component, as it went through several
changes in the past few months.

--Vijay

On Sun, Oct 25, 2015 at 3:45 PM, Yiming Wang <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/nnet3/components.py
> <https://github.com/freewym/kaldi/pull/1#discussion_r42949304>:
>
> > @@ -347,4 +347,164 @@ def AddClstmLayer(config_lines,
> >              }
> >
> >
> > +def AddBlstmLayer(config_lines,
>
> Added. One thing:
>
> in your example:
> y(t) = Wyr * r(t) + Wyp * p(t) + by
>
> But according to the existing AddLstmLayer, it seems that it is
> y(t) = [r_forward(t); p_forward(t)]
> i.e. there is no further affine transforms Wyr and Wyp on r(t) and p(t)
> respectively.
>
> SInce my Blstm is built on the the exsisting Lstm, I also wrote y(t) in
> the Append form.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freewym/kaldi/pull/1/files#r42949304>.
>